dref/imminent_utils.py

```diff
@@ -107,61 +104,54 @@ def parse_type_of_onset(type):
     paragraph_element2 = paragraph2._element.xpath('.//w:t')
-    data['title'] = paragraph_element2[0].text
-    table = document.tables[0]
-    table_row_one_column_zero = table.cell(1 , 0)._tc.xpath('.//w:t')
-    data['appeal_code'] = table_row_one_column_zero[0].text
-    table_row_one_column_one = table.cell(1, 1)._tc.xpath('.//w:t')
-    data['amount_requested'] = parse_string_to_int(table_row_one_column_one[1].text)
-    table_row_one_colum_two = table.cell(1, 2)._tc.xpath('.//w:t')
-    data['disaster_category'] = parse_disaster_category(table_row_one_colum_two[0].text)
-    table_row_one_column_four = table.cell(1, 4)._tc.xpath('.//w:t')
-    data['disaster_type'] = parse_disaster_type(table_row_one_column_four[0].text)
-    table_row_three_column_one = table.cell(3, 0)._tc.xpath('.//w:t')
-    data['glide_code'] = table_row_three_column_one[0].text
-    table_row_three_column_one = table.cell(3, 1)._tc.xpath('.//w:t')
-    data['num_affected'] = parse_string_to_int(table_row_three_column_one[0].text)
-    table_row_three_column_two = table.cell(3, 2)._tc.xpath('.//w:t')
-    data['num_assisted'] = parse_string_to_int(table_row_three_column_two[0].text)
-    table_row_five_column_zero = table.cell(5, 0)._tc.xpath('.//w:t')
-    data['type_of_onset'] = parse_type_of_onset(table_row_five_column_zero[0].text)
-    table_row_five_column_one = table.cell(5, 1)._tc.xpath('.//w:t')
-    data['date_of_approval'] = is_valid_date(table_row_five_column_one[0].text)
-    table_row_five_column_two = table.cell(5, 2)._tc.xpath('.//w:t')
-    data['end_date'] = is_valid_date(table_row_five_column_two[0].text)
-    table_row_five_colum_three = table.cell(5, 3)._tc.xpath('.//w:t')
-    data['operation_timeframe'] = int(table_row_five_colum_three[0].text)
-    table_row_six_colum_zero = table.cell(6, 0)._tc.xpath('.//w:t')
-    data['country'] = Country.objects.get(name__icontains=table_row_six_colum_zero[1].text)
+    data['title'] = get_text_or_null(paragraph_element2)
+    if not data['title']:
+        raise serializers.ValidationError('A title is required')
+
+    cells = get_table_cells(0)
+    data['appeal_code'] = cells(1, 0)
+    data['amount_requested'] = parse_string_to_int(cells(1,1,1))
+    data['disaster_category'] = parse_disaster_category(cells(1, 2))
+    data['disaster_type'] = parse_disaster_type(cells(1,4))
+    data['glide_code'] = cells(3, 0)
+    data['num_affected'] = parse_string_to_int(cells(3,1))
+    data['num_assisted'] = parse_string_to_int(cells(3,2))
+    data['type_of_onset'] = parse_type_of_onset(cells(5,0))
+    data['date_of_approval'] = parse_date(cells(5,1))
+    data['end_date'] = parse_date(cells(5,2))
+    data['operation_timeframe'] = parse_int(cells(5,3))
+    data['country'] = Country.objects.filter(name__icontains=cells(6,0,1)).first()
+    if data['country'] is None:
+        raise serializers.ValidationError('A valid country is required')
+
     paragraph6 = document.paragraphs[7]._element.xpath('.//w:t')
```

Comment:
Access to a protected member _element of a class (not a bug, but if you can smooth it, do it please).
Later the same.